
The -M* options that control dependency file creation do not affact the output
and especially -MF is used with randomized filenames by libtool and thus
invalidate the hash unnecessarily.

  Rene Rebe <rene@exactcode.de>

--- ccache-2.4/ccache.c	2004-09-13 12:38:30.000000000 +0200
+++ ccache-2.4-improved/ccache.c	2004-12-29 22:38:55.984560272 +0100
@@ -291,6 +291,7 @@
 			    strcmp(args->argv[i], "-include") == 0 ||
 			    strcmp(args->argv[i], "-L") == 0 ||
 			    strcmp(args->argv[i], "-D") == 0 ||
+			    strcmp(args->argv[i], "-MF") == 0 ||
 			    strcmp(args->argv[i], "-idirafter") == 0 ||
 			    strcmp(args->argv[i], "-isystem") == 0) {
 				i++;
@@ -300,6 +301,9 @@
 		if (strncmp(args->argv[i], "-I", 2) == 0 ||
 		    strncmp(args->argv[i], "-L", 2) == 0 ||
 		    strncmp(args->argv[i], "-D", 2) == 0 ||
+		    strncmp(args->argv[i], "-MD", 2) == 0 ||
+		    strncmp(args->argv[i], "-MP", 2) == 0 ||
+		    strncmp(args->argv[i], "-MF", 2) == 0 ||
 		    strncmp(args->argv[i], "-idirafter", 10) == 0 ||
 		    strncmp(args->argv[i], "-isystem", 8) == 0) {
 			continue;
